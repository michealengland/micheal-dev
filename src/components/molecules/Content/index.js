import PropTypes from 'prop-types'
import React from 'react'
import createMarkup from '@utilities/createMarkup'
import Heading from '@components/atoms/Heading'

export default function Content({content, title, titleTag}) {
  return (
    <section>
      {title && <Heading tag={titleTag}>{title}</Heading>}
      {content && <div dangerouslySetInnerHTML={ createMarkup(content) } />}
    </section>
  )
}

Content.propTypes = {
  content: PropTypes.string,
  title: PropTypes.string,
  titleTag: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6'])
}

Content.defaultProps = {
  titleTag: 'h2'
}
