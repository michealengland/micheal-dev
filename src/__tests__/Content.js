import React from 'react'
import renderer from 'react-test-renderer'
import Content from '@components/molecules/Content'
describe('Content', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Content
          title={'Content title.'}
          titleTag="h3"
          content="Text content..."
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
