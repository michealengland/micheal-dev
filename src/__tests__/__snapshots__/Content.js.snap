// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<section
  className="content"
>
  <h3
    className="contentHeading"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Content title.",
      }
    }
  />
  <div
    className="text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Text content...",
      }
    }
  />
</section>
`;
